cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -135,15 +139,20 @@
   private final NamespaceQueryAdmin namespaceQueryAdmin;
   private final CapabilityReader capabilityReader;
   private final File tmpDir;
+  private final AccessEnforcer accessEnforcer;
+  private final AuthenticationContext authenticationContext;
 
   @Inject
   ArtifactHttpHandler(CConfiguration cConf, ArtifactRepository artifactRepository,
-                      NamespaceQueryAdmin namespaceQueryAdmin, CapabilityReader capabilityReader) {
+                      NamespaceQueryAdmin namespaceQueryAdmin, CapabilityReader capabilityReader,
+                      AccessEnforcer accessEnforcer, AuthenticationContext authenticationContext) {
```

Comment:
nit: Consider using ContextAccessEnforcer to save you from passing in both accessEnforcer and authenticationContext.

Reply:
Let me do in a followup PR to fix all places